torch/onnx/verification.py

```diff
@@ -136,136 +72,268 @@ def _run_ort(ort_sess, inputs):
+
+    ONNXRuntime is used for model execution backend for ONNX model.
+
+    Raise:
+        AssertionError: if outputs from ONNX model and PyTorch model are not
+            equal up to specified precision.
+    """
+
+    def compare_ort_pytorch_model_with_input(input_args, input_kwargs):
+        pt_args, pt_kwargs = _prepare_input_for_pytorch(input_args, input_kwargs)
+        # TODO: remove this and treat mutating model separately. See #77679
+        model_copy = _try_clone_model(model)
+        pt_outs = model_copy(*pt_args, **pt_kwargs)
+
+        ort_inputs = _prepare_input_for_ort(
+            input_args, input_kwargs, remained_onnx_input_idx, flatten
+        )
+        ort_outs = _run_ort(ort_session, ort_inputs)
+
+        _compare_ort_pytorch_outputs(ort_outs, pt_outs, rtol, atol)
+
+    compare_ort_pytorch_model_with_input(input_args, input_kwargs)
+
+    if additional_test_inputs:
+        for test_input_args in additional_test_inputs:
+            compare_ort_pytorch_model_with_input(test_input_args, {})
 
 
 def verify(
     model: Union[torch.nn.Module, torch.jit.ScriptModule],
-    input=None,
-    batch_size=2,
-    rtol=0.001,
-    atol=1e-7,
+    input_args,
+    input_kwargs=None,
     do_constant_folding=True,
     dynamic_axes=None,
-    test_with_inputs=None,
     input_names=None,
     output_names=None,
-    fixed_batch_size=False,
-    dict_check=True,
     training=None,
+    opset_version=None,
+    keep_initializers_as_inputs=True,
+    verbose=False,
+    fixed_batch_size=False,
+    use_external_data=False,
+    additional_test_inputs=None,
     remained_onnx_input_idx=None,
     flatten=True,
-    verbose=False,
     ort_providers=_ORT_PROVIDERS,
-    ort_optim_on=True,
-    use_external_data=False,
-    opset_version: Optional[int] = None,
-    keep_initializers_as_inputs: bool = True,
+    rtol=0.001,
+    atol=1e-7,
+    **kwargs,
 ):
-    """Verify model export to ONNX with ONNXRuntime."""
+    """Verify model export to ONNX with ONNXRuntime.
+
+    Args:
+        model (torch.nn.Module or torch.jit.ScriptModule): See :func:`torch.onnx.export`.
+        input_args (tuple): See :func:`torch.onnx.export`.
+        input_kwargs (dict): See :func:`torch.onnx.export`.
+        do_constant_folding (bool, optional): See :func:`torch.onnx.export`.
+        dynamic_axes (dict, optional): See :func:`torch.onnx.export`.
+        input_names (list, optional): See :func:`torch.onnx.export`.
+        output_names (list, optional): See :func:`torch.onnx.export`.
+        training (bool, optional): See :func:`torch.onnx.export`.
+        opset_version (int, optional): See :func:`torch.onnx.export`.
+        keep_initializers_as_inputs (bool, optional): See :func:`torch.onnx.export`.
+        verbose (bool, optional): See :func:`torch.onnx.export`.
+        fixed_batch_size (bool, optional): Legacy argument, used only by rnn test cases.
+        use_external_data (bool, optional): Explicitly specify whether to export the
```

Comment:
This parameter has been removed from onnx.export() function. I think we can remove it here too.

Reply:
The purpose is slightly different. This affects how model file path is selected.